src/dbnode/persist/fs/read.go

```diff
@@ -465,6 +452,28 @@ func (r *reader) Read() (ident.ID, ident.TagIterator, checked.Bytes, uint32, err
 	return id, tags, data, uint32(entry.DataChecksum), nil
 }
 
+func (r *reader) StreamingReadMetadata() (ident.BytesID, ts.EncodedTags, int, uint32, error) {
+	if !r.streamingEnabled {
+		return nil, nil, 0, 0, errStreamingRequired
+	}
+
+	if r.metadataRead >= r.entries {
+		return nil, nil, 0, 0, io.EOF
+	}
+
+	entry, err := r.decoder.DecodeIndexEntry(nil)
+	if err != nil {
+		return nil, nil, 0, 0, err
+	}
+
+	r.streamingID = append(r.streamingID[:0], entry.ID...)
+	r.streamingTags = append(r.streamingTags[:0], entry.EncodedTags...)
```

Comment:
Will these compete with `StreamingRead` also mutating these fields?

Reply:
No. At least for now (and most probably in the future) data and metadata reads are mutually exclusive (like it was before the introduction of streaming). This is documented in the method comments on the interface.